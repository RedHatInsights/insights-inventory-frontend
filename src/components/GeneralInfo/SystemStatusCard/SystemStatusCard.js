/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import LoadingCard from '../LoadingCard';
import DateFormat from '@redhat-cloud-services/frontend-components/DateFormat';
import { systemStatus } from '../selectors';

const SystemStatusCardCore = ({
  detailLoaded,
  hasState,
  hasRegistered,
  hasLastCheckIn,
  hasRHC,
  entity,
  systemProfile,
  systemStatus,
}) => (
  <LoadingCard
    title="System status"
    isLoading={!detailLoaded}
    items={[
      ...(hasState
        ? [
            {
              title: 'Current state',
              value: systemStatus.stale ? 'Stale' : 'Active',
            },
          ]
        : []),
      ...(hasRegistered
        ? [
            {
              title: 'Registered',
              value: entity && (
                <DateFormat date={entity.created} type="exact" />
              ),
            },
          ]
        : []),
      ...(hasLastCheckIn
        ? [
            {
              title: 'Last upload',
              value: entity && (
                <DateFormat date={entity.updated} type="exact" />
              ),
            },
          ]
        : []),
      ...(hasRHC
        ? [
            {
              title: 'RHC',
              value: systemProfile?.rhc_client_id
                ? 'Connected'
                : 'Not available',
            },
          ]
        : []),
    ]}
  />
);

SystemStatusCardCore.propTypes = {
  detailLoaded: PropTypes.bool,
  entity: PropTypes.shape({
    updated: PropTypes.string,
    created: PropTypes.string,
  }),
  systemProfile: PropTypes.shape({
    rhc_client_id: PropTypes.string,
  }),
  systemStatus: PropTypes.object,
  handleClick: PropTypes.func,
  hasState: PropTypes.bool,
  hasLastCheckIn: PropTypes.bool,
  hasRegistered: PropTypes.bool,
  hasRHC: PropTypes.bool,
};
SystemStatusCardCore.defaultProps = {
  detailLoaded: false,
  systemStatus: {},
  handleClick: () => undefined,
  hasState: true,
  hasLastCheckIn: true,
  hasRegistered: true,
  hasRHC: true,
};

export const SystemStatusCard = connect(
  ({ entityDetails: { entity }, systemProfileStore: { systemProfile } }) => ({
    entity,
    systemProfile,
    detailLoaded: systemProfile?.loaded,
    systemStatus: systemStatus(entity),
  })
)(SystemStatusCardCore);

SystemStatusCard.propTypes = SystemStatusCardCore.propTypes;
SystemStatusCard.defaultProps = SystemStatusCardCore.defaultProps;

export default SystemStatusCard;
