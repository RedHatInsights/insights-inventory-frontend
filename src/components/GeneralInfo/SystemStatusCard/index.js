export { default } from './SystemStatusCard';
export * from './SystemStatusCard';
