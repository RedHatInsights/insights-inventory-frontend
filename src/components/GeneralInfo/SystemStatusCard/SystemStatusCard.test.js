import React from 'react';
import SystemStatusCard from './SystemStatusCard';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'localhost:3000/example/path',
  }),
  useHistory: () => ({
    push: () => undefined,
  }),
}));

describe('SystemStatusCard', () => {
  let initialState;
  let mockStore;

  beforeEach(() => {
    mockStore = configureStore();
    initialState = {
      entityDetails: {
        entity: {
          updated: '6/01/2014',
          created: '04/01/2014',
        },
      },
      systemProfileStore: {
        systemProfile: {
          loaded: true,
          rhc_client_id: ['1234'],
        },
      },
    };
  });

  it('should render correctly - no data', () => {
    const store = mockStore({ systemProfileStore: {}, entityDetails: {} });
    const view = render(<SystemStatusCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  it('should render correctly with data', () => {
    const store = mockStore(initialState);
    const view = render(<SystemStatusCard store={store} />);
    expect(view.asFragment()).toMatchSnapshot();
  });

  ['hasState', 'hasLastCheckIn', 'hasRegistered', 'hasRHC'].map((item) =>
    it(`should not render ${item}`, () => {
      const store = mockStore(initialState);
      const view = render(
        <SystemStatusCard store={store} {...{ [item]: false }} />
      );
      expect(view.asFragment()).toMatchSnapshot();
    })
  );
});
