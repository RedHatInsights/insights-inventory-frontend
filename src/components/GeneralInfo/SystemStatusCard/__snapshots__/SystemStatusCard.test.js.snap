// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemStatusCard should not render hasLastCheckIn 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-4"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-4"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            System status
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Current state
            </dt>
            <dd
              class=""
            >
              Active
            </dd>
            <dt
              class=""
            >
              Registered
            </dt>
            <dd
              class=""
            >
              01 Apr 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              RHC
            </dt>
            <dd
              class=""
            >
              Connected
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`SystemStatusCard should not render hasRHC 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-6"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-6"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            System status
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Current state
            </dt>
            <dd
              class=""
            >
              Active
            </dd>
            <dt
              class=""
            >
              Registered
            </dt>
            <dd
              class=""
            >
              01 Apr 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              Last upload
            </dt>
            <dd
              class=""
            >
              01 Jun 2014 00:00 UTC
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`SystemStatusCard should not render hasRegistered 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-5"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-5"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            System status
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Current state
            </dt>
            <dd
              class=""
            >
              Active
            </dd>
            <dt
              class=""
            >
              Last upload
            </dt>
            <dd
              class=""
            >
              01 Jun 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              RHC
            </dt>
            <dd
              class=""
            >
              Connected
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`SystemStatusCard should not render hasState 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-3"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-3"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            System status
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Registered
            </dt>
            <dd
              class=""
            >
              01 Apr 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              Last upload
            </dt>
            <dd
              class=""
            >
              01 Jun 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              RHC
            </dt>
            <dd
              class=""
            >
              Connected
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`SystemStatusCard should render correctly - no data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-1"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-1"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            System status
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Current state
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Registered
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              Last upload
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
            <dt
              class=""
            >
              RHC
            </dt>
            <dd
              class=""
            >
              <div
                class="pf-c-skeleton ins-c-skeleton ins-c-skeleton__sm"
              >
                <span
                  class="pf-u-screen-reader"
                />
              </div>
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`SystemStatusCard should render correctly with data 1`] = `
<article
  class="pf-c-card"
  data-ouia-component-id="OUIA-Generated-Card-2"
  data-ouia-component-type="PF4/Card"
  data-ouia-safe="true"
  id=""
>
  <div
    class="pf-c-card__body"
  >
    <div
      class="pf-l-stack pf-m-gutter"
    >
      <div
        class="pf-l-stack__item"
      >
        <div
          class="pf-c-content"
        >
          <h1
            class=""
            data-ouia-component-id="OUIA-Generated-Text-2"
            data-ouia-component-type="PF4/Text"
            data-ouia-safe="true"
            data-pf-content="true"
          >
            System status
          </h1>
        </div>
      </div>
      <div
        class="pf-l-stack__item pf-m-fill"
      >
        <div
          class="pf-c-content"
        >
          <dl
            class=""
          >
            <dt
              class=""
            >
              Current state
            </dt>
            <dd
              class=""
            >
              Active
            </dd>
            <dt
              class=""
            >
              Registered
            </dt>
            <dd
              class=""
            >
              01 Apr 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              Last upload
            </dt>
            <dd
              class=""
            >
              01 Jun 2014 00:00 UTC
            </dd>
            <dt
              class=""
            >
              RHC
            </dt>
            <dd
              class=""
            >
              Connected
            </dd>
          </dl>
        </div>
      </div>
    </div>
  </div>
</article>
`;
